### PR TITLE
Fix: rendezvous: do not reject incoming packet with id=0 when rendezvous

### LINKED ISSUE #2539: Rendezvous Connection in the Non Blocking Mode
There appears to be a bug in the rendezvous handshake flow of SRT. Checked with the latest master and SRT v1.4.2. I expect earlier versions are affected as well.
The bug is not critical because a connection still can be established. Just the flow is not totally correct and in a certain case establishing a connection would take longer: ~(2,5×RTT + 250 ms) instead of 1,5×RTT.
On the left is the expected (correct) rendezvous HS flow that happens in the blocking mode of SRT receiving ([SRTO_RCVSYN](https://github.com/Haivision/srt/blob/master/docs/API/API-socket-options.md#SRTO_RCVSYN)=true).
Connection is established in 1,5×RTT (from the time both peers are up and running).
On the right is the (incorrect) rendezvous HS flow that happens in the non-blocking mode of SRT receiving ([SRTO_RCVSYN](https://github.com/Haivision/srt/blob/master/docs/API/API-socket-options.md#SRTO_RCVSYN)=false).
Connection is established in (2,5×RTT + 250ms) from the time both peers are up and running.
It is however not a critical issue, because the connection is still established.

image (1)

### Reason for WAVEAHAND instead of CONCLUSION REQ

The initial WAVEAHAND does not have a DST SRT Socket ID, [the following condition](https://github.com/Haivision/srt/blob/v1.5.1/srtcore/queue.cpp#L911) is true:
```c++
        if (i->id != dest_id)
        {
            read_st = RST_AGAIN;
            conn_st = CONN_AGAIN;
        }
```

Setting `conn_st = CONN_AGAIN` forces another `WAVEAHAND` instead of `CONCLUSION_REQ`.

### PR DESCRIPTION
**Pull request by ethouris**
Fixes #2539

### CODE REVIEW

File: srtcore/queue.cpp [hunk 1]
```diff
@@ -897,10 +897,26 @@ void srt::CRendezvousQueue::updateConnStatus(EReadStatus rst, EConnectStatus cst
...
+            {
+                HLOGC(cnlog.Debug, log << "updateConnStatus: cst=" << ConnectStatusStr(cst) << " but for RID @" << i->id
+                        << " dest_id=@" << dest_id << " - resetting to AGAIN");
+
+                read_st = RST_AGAIN;
+                conn_st = CONN_AGAIN;
+            }
+            else
+            {
+                HLOGC(cnlog.Debug, log << "updateConnStatus: cst=" << ConnectStatusStr(cst) << " and update came for @"
```
Comment (maxsharabayko) [inline srtcore/queue.cpp:912]:
```suggestion
                HLOGC(cnlog.Debug, log << "updateConnStatus: cst=" << ConnectStatusStr(cst) << " for @"
```

**Review by maxsharabayko — COMMENTED** [review]
Minor suggestion.

**Comment by maxsharabayko** [comment]
Wow, segmentation fault. Not sure if it is related to the changes in the PR.
```shell
[ RUN      ] TestEnforcedEncryption.CASE_D_4_NonBlocking_Enforced_Off_Off_Pwd_None_Set
09:05:31.484224/SRT:RcvQ:w75!W:SRT.cn: @254575997: HS EXT: Agent declares encryption, but Peer does not (Agent can still receive unencrypted packets from Peer).
09:05:31.485487/SRT:RcvQ:w75!W:SRT.cn: @254575997: createSrtHandshake: Agent has PW, but Peer sent no KMREQ. Sending error KMRSP response
Epoll wait result: 1 FOUND: @254575998 in [R]
09:05:31.485936/SRT:RcvQ:w76!W:SRT.cn: @254575999: HS KMREQ: Peer declares encryption, but agent does not - still allowing connection.
09:05:31.485958/SRT:RcvQ:w76!W:SRT.cn: processSrtMsg_KMRSP: received failure report. STATE: NOSECRET
ACCEPT: done, result=254575997
[T] ACCEPT SUCCEEDED: @254575997
/home/travis/.travis/functions: line 109:  7067 Segmentation fault      ./test-srt --gtest_filter="-$TESTS_IPv6"
```

**Comment by ethouris** [comment]
Well, that's kinda weird. I tried this now and no problems are reported on the local machine. I also haven't changed anything that would influence this flow. This might influence the handshake processing on the listener side, but here in the test the segfault is reported after the accept call has succeeded on both endpoints, so it's after the handshake. Worth researching, but I doubt we can retrieve enough data from Travis.

**Pull request merged**